packages/kilt-did/README.md

```diff
@@ -3,6 +3,320 @@
+const keystore = new DemoKeystore()
+
+// Generate seed for the authentication key.
+const authenticationSeed = '0x123456789'
+
+// Ask the keystore to generate a new keypar to use for authentication.
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate the DID-signed creation extrinsic.
+// The extrinsic is unsigned and contains the DID creation operation signed with the DID authentication key.
+const { submittable, did } = await DidUtils.writeDidfromPublicKeys(keystore, {
+  [KeyRelationship.authentication]: {
+    publicKey: authenticationKeyPublicDetails.publicKey,
+    type: authenticationKeyPublicDetails.alg,
+  },
+})
+// Will print `did:kilt:4sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar`.
+console.log(did)
+
+// Submit the DID creation tx to the KILT blockchain after signing it with the KILT account specified.
+await BlockchainUtils.signAndSubmitTx(submittable, aliceKiltAccount, {
+  resolveOn: BlockchainUtils.IS_IN_BLOCK,
+})
+
+const fullDid = (await DefaultResolver.resolveDoc(did)) as FullDidDetails
+```
+
+If additional keys and external services are to be specified, then they can be included in the DID create operation.
+
+```typescript
+await Kilt.init({ address: 'ws://127.0.0.1:9944' })
+
+const aliceKiltAccount = new Keyring({
+  type: 'ed25519',
+  ss58Format: 38,
+}).createFromUri('//Alice')
+
+const keystore = new DemoKeystore()
+
+const authenticationSeed = '0x123456789'
+
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  seed: authenticationSeed,
+  alg: SigningAlgorithms.Ed25519,
+})
+
+// Generate seed for the encryption key.
+const encryptionSeed = '0x987654321'
+
+// Ask the keystore to generate a new keypar to use for encryption.
+const encryptionKeyPublicDetails = await keystore.generateKeypair({
+  seed: encryptionSeed,
+  alg: EncryptionAlgorithms.NaclBox,
+})
+
+// Define the set of URLs pointing to the external services exposed by the DID subject.
+// All URLs must point to the same resource, which must match the provided hash.
+const serviceEndpoints: EndpointData = {
+  contentHash: 'external-endpoints-content-hash00',
+  contentType: 'application/json',
+  urls: [
+    'https://myendpoint.io?endpoint_id=my_endpoint',
+    'ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu',
+    'ftp://myendpoint.io/endpoints/example.json',
+  ],
+}
+
+// Generate the DID-signed creation extrinsic with the provided keys and service endpoints.
+const { submittable, did } = await DidUtils.writeDidfromPublicKeys(
+  keystore,
+  {
+    [KeyRelationship.authentication]: {
+      publicKey: authenticationKeyPublicDetails.publicKey,
+      type: authenticationKeyPublicDetails.alg,
+    },
+    [KeyRelationship.keyAgreement]: {
+      publicKey: encryptionKeyPublicDetails.publicKey,
+      // The chain-supported type is called differently than the type supported in the keystore
+      type: 'x25519',
```

Comment:
If we control both the chain and the store, could we align the names? Or at least add a synonym?

Reply:
As the keystore is developed and managed by developers, I am exposing just a utility function that can be called on the DemoKeystore to get the right key type. Otherwise semantically the keystore needs more information about a key than the chain does, so it is not possible to call them the same.

Reply:
Since it would be irrelevant for the actual production code, I suggest either removing this comment or modifying it to make clear this is only about the demo keystore
```suggestion
      // The chain-supported type is called differently than the type supported in the demo keystore
      type: 'x25519',
```